docs/man/man1/activator.1
.\" Automatically generated by Pod::Man 2.23 (Pod::Simple 3.16)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.ie \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.el \{\
.    de IX
..
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "ACTIVATOR 1"
.TH ACTIVATOR 1 "2011-08-20" "perl v5.12.3" "Entropy"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
activator \- Official Sabayon Linux Server\-Side Repositories Management tool.
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBactivator\fR [action] [basic_options] [options] [atom | package_file | \f(CW@set\fR] ...
.PP
\&\fBactivator\fR [action] [basic_options] [options] ...
.PP
\&\fBactivator\fR \-\-help
.PP
\&\fBactivator\fR \-\-version
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBactivator\fR is a server-side repository administration tool used to handle
remote Entropy repository mirrors.
.SH "BASIC_OPTIONS"
.IX Header "BASIC_OPTIONS"
.IP "\fB\-\-help (\-h)\fR" 4
.IX Item "--help (-h)"
this output
.IP "\fB\-\-version\fR" 4
.IX Item "--version"
print version
.IP "\fB\-\-nocolor\fR" 4
.IX Item "--nocolor"
disable colorized output
.IP "\fB\-\-color\fR" 4
.IX Item "--color"
force colorized output
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fBsync\fR" 4
.IX Item "sync"
sync packages, database and also do some tidy
.RS 4
.IP "\fB\-\-branch=<branch\fR>" 4
.IX Item "--branch=<branch>"
choose on what branch operating
.IP "\fB\-\-noask\fR" 4
.IX Item "--noask"
do not ask anything except critical things
.IP "\fB\-\-syncall\fR" 4
.IX Item "--syncall"
sync all the configured repositories
.RE
.RS 4
.RE
.IP "\fBsyncas <repository\fR>" 4
.IX Item "syncas <repository>"
same as sync, but put repository database \s-1ONLY\s0 into a dummy repository
.IP "\fBtidy\fR" 4
.IX Item "tidy"
remove binary packages not in repositories and expired
.IP "\fBpackages\fR" 4
.IX Item "packages"
package repositories handling functions
.RS 4
.IP "\fBsync\fR" 4
.IX Item "sync"
sync package repositories across primary mirrors
.RS 4
.IP "\fB\-\-ask\fR" 4
.IX Item "--ask"
ask before making any changes
.IP "\fB\-\-pretend\fR" 4
.IX Item "--pretend"
only show what would be done
.IP "\fB\-\-syncall\fR" 4
.IX Item "--syncall"
sync all the configured repositories
.IP "\fB\-\-do\-packages\-check\fR" 4
.IX Item "--do-packages-check"
also verify packages integrity
.RE
.RS 4
.RE
.RE
.RS 4
.RE
.IP "\fBrepo\fR" 4
.IX Item "repo"
repository handling functions
.RS 4
.IP "\fBsync\fR" 4
.IX Item "sync"
sync the current repository database across primary mirrors
.RS 4
.IP "\fB\-\-syncall\fR" 4
.IX Item "--syncall"
sync all the configured repositories
.RE
.RS 4
.RE
.IP "\fBsyncas <repository\fR>" 4
.IX Item "syncas <repository>"
same as sync, but put repository database \s-1ONLY\s0 into a dummy repository
.IP "\fBvacuum\fR" 4
.IX Item "vacuum"
clean unavailable packages from mirrors (similar to tidy, but more nazi)
.RS 4
.IP "\fB\-\-days=<days\fR>" 4
.IX Item "--days=<days>"
expiration days [default is: 0]
.RE
.RS 4
.RE
.IP "\fBlock\fR" 4
.IX Item "lock"
lock the current repository database (server-side)
.IP "\fBunlock\fR" 4
.IX Item "unlock"
unlock the current repository database (server-side)
.IP "\fBdownload-lock\fR" 4
.IX Item "download-lock"
lock the current repository database (client-side)
.IP "\fBdownload-unlock\fR" 4
.IX Item "download-unlock"
unlock the current repository database (client-side)
.IP "\fBlock-status\fR" 4
.IX Item "lock-status"
show current lock status
.RE
.RS 4
.RE
.IP "\fBnotice\fR" 4
.IX Item "notice"
notice board handling functions
.RS 4
.IP "\fBadd\fR" 4
.IX Item "add"
add a news item to the notice board
.IP "\fBremove\fR" 4
.IX Item "remove"
remove a news item from the notice board
.IP "\fBread\fR" 4
.IX Item "read"
read the current notice board
.RE
.RS 4
.RE
.SH "ENVIRONMENT"
.IX Header "ENVIRONMENT"
\&\fB\s-1ETP_REPO\s0\fR=valid repository id: on-the-fly switch default Entropy repository.
.PP
\&\fB\s-1ETP_EXPIRATION_DAYS\s0\fR=valid number: on-the-fly removed packages expiration days
setting.
.SH "BUGS"
.IX Header "BUGS"
Please report bugs to http://bugs.sabayonlinux.org.
.SH "EXIT STATUS"
.IX Header "EXIT STATUS"
\&\fBactivator\fR returns a zero exit status if the called command succeeded. Non zero
is returned in case of failure.
.SH "AUTHOR"
.IX Header "AUTHOR"
Fabio Erculiani <lxnay@sabayon.org>
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\&\fIreagent\fR\|(1), \fIequo\fR\|(1)
.SH "POD ERRORS"
.IX Header "POD ERRORS"
Hey! \fBThe above document had some coding errors, which are explained below:\fR
.IP "Around line 197:" 4
.IX Item "Around line 197:"
You forgot a '=back' before '=head1'


docs/ChangeLog
commit 266fc224abe3f00312bcab4a878a01a391bd41c9
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Thu Aug 25 19:36:52 2011 +0200

    Release Entropy 1.0_rc44

commit 6059629452d2898b4f66b5a632f3d7834fe05d10
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Thu Aug 25 19:36:00 2011 +0200

    [entropy.spm] PortagePlugin.add_installed_package: chmod vardb entry to 755 as Portage does

commit 949f2ab3f3ed1c844a02670fe11b65fb6a284031
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Thu Aug 25 18:39:04 2011 +0200

    [entropy.server] add PKG_KEYWORDS variable to those exposed to metadata qa check hook

commit b0a2bbf4fe6d453a63b35be3b17104a8121ccefd
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Thu Aug 25 09:52:42 2011 +0200

    [entropy.server.mirrors] do not handle .asc files anymore in tidy_mirrors()

commit 7ead04fa0180e0d79a66804ecd6c162876a72183
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Thu Aug 25 09:52:17 2011 +0200

    [kernel-switcher] catch KeyboardInterrupt and don't throw traceback to user

commit 1d29a15e056947066157571d8e4d7f37c0415279
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Aug 24 13:45:11 2011 +0200

    [matter] remove temp file in buildfail-tinderbox.sh

commit 2fca78342580aa3897b7b217f9ae5572f2d8ecf2
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Aug 24 13:43:35 2011 +0200

    [matter] add simple buildfail example script

commit 7b5d009149aa65aa92d2351f9e6f65fa09d6ad10
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Aug 24 12:32:35 2011 +0200

    [matter] add bugzilla submit buildfail hook

commit 3a9cbb6341e55ab803c86523770241861a16e5dc
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Aug 24 09:57:58 2011 +0200

    [matter] add "buildfail" directive support

commit 9b22ff56136593fed1e3e5d687c2ac157c44140f
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Aug 24 08:48:29 2011 +0200

    [matter] add support for "not-installed" directive

commit 32c6dc7a011d3c258a2d8586d41e35ec061c3abe
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Aug 24 08:25:43 2011 +0200

    [matter] add FEATURES="split-log" and CMAKE_NO_COLOR="yes" to builder env

commit ca3fa9daba03760cfedee54ce8b3cb2955605124
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:34:23 2011 +0200

    [entropy] add .gitattributes file, exclude test suite from archive for now (takes too much space)

commit 698129ba8460e7d27d6662c6889387404ee66c25
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:23:04 2011 +0200

    Tagging Entropy version 1.0_rc43

commit 9366621d90e600874cfa3c1a284be72503cecd11
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:23:02 2011 +0200

    Release Entropy 1.0_rc43

commit 13c3b7513d5c2664230e5effe8be8522635d7652
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:17:48 2011 +0200

    [entropy.server.mirrors] implement herustic able to use remote file copy when possible
    
    When dealing with multiple repositories, it is possible that a package file
    gets uploaded multiple times when moved from a repository to another.
    In this scenario, there is a waste of upload bandwidth. This commit
    implements a simple herustic that is able to determine if it's possible
    to copy the file remotely instead of uploading it, if and only if:
    
    - package file with the same name is already available on another
      tracked repository sharing the same mirror URI
    - local package and remote package have the same md5 hash

commit a61f6b99cc567b3aec8a9db36f8e454e6e34cd47
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:16:23 2011 +0200

    [entropy.transceivers] EntropyFtpUriHandler: fix undefined reference

commit 1446115d33eae2e1d9f644d211596e30fcfdbe94
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:13:56 2011 +0200

    [entropy.transceivers] implement remote copy method for all transceivers (similar to rename())

commit 9c55829d00cfe6abe3e61002751432e01cf1671f
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:11:22 2011 +0200

    [entropy.transceivers] EntropyFileUriHandler.rename: fix typo

commit 9bb680ea67ea9432ca9636675c3aabd609f24726
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 14:02:38 2011 +0200

    [activator] tweak commit message file name

commit 0210ee7b97a86dcd513c06fe288a4ba138007e7c
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 11:13:45 2011 +0200

    [entropy.server.transceivers] code style update

commit 247e6fcc102d89158fe36aaefc9bb07cf2d7177a
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sun Aug 21 10:53:33 2011 +0200

    [services] call env-update with full path

commit d490a84b8429e7d712e8d395cf0ec8284e150894
Author: Sławomir Nizio <slawomir.nizio@sabayon.org>
Date:   Sat Aug 20 21:17:36 2011 +0200

    [po] update Polish translation

commit bc4e780274f6383678849de19e7022a30a04abac
Author: Sławomir Nizio <slawomir.nizio@sabayon.org>
Date:   Sat Aug 20 21:15:09 2011 +0200

    [misc] fix some typos in equo and manual pages

commit a4bda31855caa62adeac8650e27f93677e4e61c2
Author: Sławomir Nizio <slawomir.nizio@sabayon.org>
Date:   Sat Aug 20 18:14:13 2011 +0200

    [man] small correction in manual page, update POD (etc.) generator

commit 97f4a3d6f9a1ce832829c9310f6d49e30340a96a
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sat Aug 20 14:01:31 2011 +0200

    Tagging Entropy version 1.0_rc42

commit 81faac522bb4d6ac94118c0e3b47a1c21b6d03ec
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sat Aug 20 14:01:23 2011 +0200

    Release Entropy 1.0_rc42

commit ff23cadda8f67da6c482891a49852798daa7cd1a
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sat Aug 20 13:44:03 2011 +0200

    [matter-scheduler] do not use "source" but rather "."

commit 15e11cc4efe7a9cb7c8d09c4122f314825f4c113
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Sat Aug 20 13:43:30 2011 +0200


server/activator.py
#!/usr/bin/python2 -O
# -*- coding: utf-8 -*-
"""

    @author: Fabio Erculiani <lxnay@sabayon.org>
    @contact: lxnay@sabayon.org
    @copyright: Fabio Erculiani
    @license: GPL-2

    B{Entropy Package Manager Server}.

"""

import os, sys
sys.path.insert(0, '../libraries')
sys.path.insert(1, '../client')
sys.path.insert(2, '../server')
sys.path.insert(3, '/usr/lib/entropy/client')
sys.path.insert(4, '/usr/lib/entropy/libraries')
sys.path.insert(5, '/usr/lib/entropy/server')
from entropy.i18n import _
import entropy.tools
from entropy.output import red, is_stdout_a_tty, nocolor, print_generic, \
    etpUi, print_error, darkgreen
from entropy.const import etpConst, const_kill_threads
from entropy.server.interfaces import Server
from text_tools import print_menu

def get_entropy_server():
    return Server(community_repo = etpConst['community']['mode'])

help_opts = [
    None,
    (0, " ~ activator ~ ", 1,
        'Entropy Package Manager - (C) %s' % (entropy.tools.get_year(),) ),
    None,
    (0, _('Basic Options'), 0, None),
    None,
    (1, '--help', 2, _('this output')),
    (1, '--version', 1, _('print version')),
    (1, '--nocolor', 1, _('disable colorized output')),
    (1, '--color', 2, _('force colorized output')),
    None,
    (0, _('Application Options'), 0, None),
    None,
    (1, 'sync', 3, _('sync packages, database and also do some tidy')),
        (2, '--branch=<branch>', 1, _('choose on what branch operating')),
        (2, '--noask', 3, _('do not ask anything except critical things')),
        (2, '--syncall', 2, _('sync all the configured repositories')),
    (1, 'syncas <repository>', 1, _('same as sync, but put repository database ONLY into a dummy repository')),
    None,
    (1, 'tidy', 3, _('remove binary packages not in repositories and expired')),
    None,
    (1, 'packages', 2, _('package repositories handling functions')),
        (2, 'sync', 3, _('sync package repositories across primary mirrors')),
            (3, '--ask', 3, _('ask before making any changes')),
            (3, '--pretend', 2, _('only show what would be done')),
            (3, '--syncall', 2, _('sync all the configured repositories')),
            (3, '--do-packages-check', 1, _('also verify packages integrity')),
    None,
    (1, 'repo', 3, _('repository handling functions')),
        (2, 'sync', 3, _('sync the current repository database across primary mirrors')),
            (3, '--syncall', 1, _('sync all the configured repositories')),
        (2, 'syncas <repository>', 1, _('same as sync, but put repository database ONLY into a dummy repository')),
        (2, 'vacuum', 3, _('clean unavailable packages from mirrors (similar to tidy, but more nazi)')),
            (3, '--days=<days>', 1, _('expiration days [default is: 0, dangerous!]')),
        (2, 'lock', 3, _('lock the current repository database (server-side)')),
        (2, 'unlock', 3, _('unlock the current repository database (server-side)')),
        (2, 'download-lock', 2, _('lock the current repository database (client-side)')),
        (2, 'download-unlock', 2, _('unlock the current repository database (client-side)')),
        (2, 'lock-status', 2, _('show current lock status')),

    None,
    (1, 'notice', 3, _('notice board handling functions')),
        (2, 'add', 2, _('add a news item to the notice board')),
        (2, 'remove', 2, _('remove a news item from the notice board')),
        (2, 'read', 2, _('read the current notice board')),
    None,
]

options = sys.argv[1:]

def handle_exception(exc_class, exc_instance, exc_tb):

    # restore original exception handler, to avoid loops
    uninstall_exception_handler()
    entropy.tools.kill_threads()

    if exc_class is KeyboardInterrupt:
        raise SystemExit(1)

    # always slap exception data (including stack content)
    entropy.tools.print_exception(tb_data = exc_tb)

    raise exc_instance

def install_exception_handler():
    sys.excepthook = handle_exception

def uninstall_exception_handler():
    sys.excepthook = sys.__excepthook__

# print version
if ("--version" in options) or ("-V" in options):
    print_generic("activator: "+etpConst['entropyversion'])
    raise SystemExit(0)

import re
opt_r = re.compile("^(\\-)([a-z]+)$")
for n in range(len(options)):
    if opt_r.match(options[n]):
        x = options[n]
        del options[n]
        options.extend(["-%s" % (d,) for d in x[1:]])

# preliminary options parsing
_options = []
force_color = False
for opt in options:
    if opt in ["--nocolor", "-N"]:
        nocolor()
    elif opt in ["--color", "-C"]:
        force_color = True
    elif opt in ["--quiet", "-q"]:
        etpUi['quiet'] = True
    elif opt in ["--verbose", "-v"]:
        etpUi['verbose'] = True
    elif opt in ["--ask", "-a"]:
        etpUi['ask'] = True
    elif opt in ["--pretend", "-p"]:
        etpUi['pretend'] = True
    else:
        _options.append(opt)
options = _options

# Check if we need to disable colors
if (not force_color) and (not is_stdout_a_tty()):
    nocolor()

# print help
if not options or ("--help" in options) or ("-h" in options):
    print_menu(help_opts)
    if len(options) < 1:
        print_error("not enough parameters")
    raise SystemExit(1)

rc = -10
if not entropy.tools.is_root():
    print_error("you must be root in order to run activator")
    rc = 2

main_cmd = options.pop(0)
install_exception_handler()

if main_cmd == "sync":
    import server_activator
    rc = server_activator.sync(options)

if main_cmd == "syncas":
    import server_activator
    rc = server_activator.syncas(options)

elif main_cmd == "tidy":
    import server_activator
    rc = server_activator.sync(options, just_tidy = True)

elif main_cmd == "repo":
    import server_activator
    rc = server_activator.repo(options)

elif main_cmd == "packages":
    import server_activator
    rc = server_activator.packages(options)

# database tool
elif main_cmd == "notice":
    import server_activator
    rc = server_activator.notice(options)

if rc == -10:
    print_menu(help_opts)
    print_error(red(" %s." % (_("Wrong parameters"),) ))
    rc = 10

uninstall_exception_handler()
const_kill_threads()
raise SystemExit(rc)
